src/main/java/Duke.java

```diff
@@ -1,10 +1,323 @@
+            break;
+        case "Deadline":
+            String by = tokens[3];
+            task = new Deadline(description, by);
+            break;
+        case "Event":
+            String at = tokens[3];
+            task = new Event(description, at);
+            break;
+        }
+        if (isDone == "1") {
+            task.setIsDone();
+        }
+        taskList.add(task);
+        taskCount += 1;
+    }
+
+    public static void printWithBorder(String line) {
+        System.out.print("_______________________________________________________________________________\n");
+        System.out.print(line + "\n");
+        System.out.print("_______________________________________________________________________________\n");
+    }
+
+    private static void inputAndExecuteCommand() {
+        String line;
+        Scanner scanner = new Scanner(System.in);
+        
+        while (true) {
+            line = scanner.nextLine();
+            String[] commandTypeAndArg = line.split(" ", 2);
+            String commandType = commandTypeAndArg[0].trim();
+            String commandArg = null;
+            if (commandTypeAndArg.length > 1) {
+                commandArg = commandTypeAndArg[1].trim();
+            }
+
+            if (commandType.equals("bye")) {
+                scanner.close();
+                return;
+            }
+            
+            try {
+                executeCommand(commandType, commandArg);
+            } catch (NullCommandArgException | InvalidCommandTimeException
+            | InvalidCommandException | InvalidTaskNumberException e) {
+                printWithBorder(e.getMessage());
+                continue;
+            }
+        }
+    }
+    
+    private static void executeCommand(String commandType, String commandArg) throws NullCommandArgException,
+    InvalidCommandTimeException, InvalidCommandException, InvalidTaskNumberException {
+        switch (commandType) {
+        case "help":
+            printWithBorder(COMMANDS);
+            break;
+        case "list":
+            listAllTasks();
+            break;
+        case "done":
+            markTaskAsDone(commandArg);
+            break;
+        case "todo":
+            addTodo(commandArg);
+            break;
+        case "deadline":
+            addDeadline(commandArg);
+            break;
+        case "event":
+            addEvent(commandArg);
+            break;
+        case "delete":
+            deleteTask(commandArg);
+            break;
+        default:
+            throw new InvalidCommandException(commandType);
+        }
+    }
+
+    private static void listAllTasks() {
+        int count = 1;
+        String listOfTasksString = "Here are the tasks in your list:";
+        for (Task task : taskList) {
+            if (task == null) {
+                break;
+            }
+            listOfTasksString += ("\n    " + Integer.toString(count) + ". " + task.toString());
+            count += 1;
```

Comment:
Is `count` an iterator variable? You could consider this format for `for` loops: See [iterator variables](https://se-education.org/guides/conventions/java/basic.html#naming).

Reply:
thank you! i will change this into a neater for loop